eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToSubmittedVariantProcessorTest.java

```diff
@@ -420,6 +420,48 @@ public void transformShortTandemRepeat() throws Exception {
+        SubSnpNoHgvs subSnpNoHgvs = new SubSnpNoHgvs(59831075L, 109775855L, "AT", "(AT)7/(AT)19", ASSEMBLY,
+                                                     BATCH_HANDLE, BATCH_NAME, CHROMOSOME, CHROMOSOME_START,
+                                                     CONTIG_NAME, CONTIG_START, DbsnpVariantType.MICROSATELLITE,
+                                                     Orientation.FORWARD, Orientation.REVERSE, Orientation.FORWARD,
+                                                     false, false, false, false, CREATED_DATE, CREATED_DATE, TAXONOMY);
+
+        List<DbsnpSubmittedVariantEntity> variants = processor.process(subSnpNoHgvs);
+        assertProcessedVariant(subSnpNoHgvs, variants.get(0), CHROMOSOME, CHROMOSOME_START, "AT", "(AT)7", false,
+                               false);
+        assertProcessedVariant(subSnpNoHgvs, variants.get(1), CHROMOSOME, CHROMOSOME_START, "AT", "(AT)19", false,
+                               false);
+    }
+
+    @Test
+    public void transformMicrosatelliteCornerCases() throws Exception {
+        // There are STR variants that don't have repeated motifs
+        SubSnpNoHgvs subSnpNoHgvs = new SubSnpNoHgvs(700214256L, 439714840L, "C", "C/AGAGTTCAAGTTGCCCTA", ASSEMBLY,
+                                                     BATCH_HANDLE, BATCH_NAME, CHROMOSOME, CHROMOSOME_START,
+                                                     CONTIG_NAME, CONTIG_START, DbsnpVariantType.MICROSATELLITE,
+                                                     Orientation.FORWARD, Orientation.FORWARD, Orientation.FORWARD,
+                                                     false, false, false, false, CREATED_DATE, CREATED_DATE, TAXONOMY);
+
+        List<DbsnpSubmittedVariantEntity> variants = processor.process(subSnpNoHgvs);
+        assertProcessedVariant(subSnpNoHgvs, variants.get(0), CHROMOSOME, CHROMOSOME_START, "C", "AGAGTTCAAGTTGCCCTA",
+                               false, true);
+
+        // there are STR variants where the alleles string is enclosed with square brackets
+        subSnpNoHgvs = new SubSnpNoHgvs(159831041L, 109548685L, "ACACACACAC", "[(GT)11/14]", ASSEMBLY, BATCH_HANDLE,
+                                        BATCH_NAME, CHROMOSOME, CHROMOSOME_START, CONTIG_NAME, CONTIG_START,
+                                        DbsnpVariantType.MICROSATELLITE, Orientation.FORWARD, Orientation.REVERSE,
+                                        Orientation.FORWARD, false, false, false, false, CREATED_DATE, CREATED_DATE,
+                                        TAXONOMY);
+
+        variants = processor.process(subSnpNoHgvs);
+        assertProcessedVariant(subSnpNoHgvs, variants.get(0), CHROMOSOME, CHROMOSOME_START, "ACACACACAC", "(AC)11",
+                               false, false);
+        assertProcessedVariant(subSnpNoHgvs, variants.get(1), CHROMOSOME, CHROMOSOME_START, "ACACACACAC", "(AC)14",
+                               false, false);
+    }
```

Comment:
Add tests that exercises the code path that throws Exceptions when the STR does not conform to the regex.

Reply:
Agreed to no throw exceptions when the STR expressions cannot be parsed